From 114ca7843aea535a1621518f4f392a40109568e6 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Fri, 26 Jun 2020 09:53:13 -0400
Subject: [PATCH 2/7] v4l2allocator: Fix data offset / bytesused size
 validation

The check was too strict causing spurious warning. Now check for <= so that 0
sized buffer do not cause a warning.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/649>

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>
Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
---
 gst-v4l2/gstv4l2allocator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gstv4l2allocator.c b/gstv4l2allocator.c
index 3fc4c39..6a53474 100644
--- a/gstv4l2allocator.c
+++ b/gstv4l2allocator.c
@@ -1505,7 +1505,7 @@ gst_v4l2_allocator_dqbuf (GstV4l2Allocator * allocator,
 
       offset = group->planes[i].data_offset;
 
-      if (group->planes[i].bytesused > group->planes[i].data_offset) {
+      if (group->planes[i].bytesused >= group->planes[i].data_offset) {
         size = group->planes[i].bytesused - group->planes[i].data_offset;
       } else {
         GST_WARNING_OBJECT (allocator, "V4L2 provided buffer has bytesused %"
-- 
2.25.1

